lib/membrane/video_compositor/video_compositor.ex

```diff
@@ -102,120 +111,224 @@ defmodule Membrane.VideoCompositor do
+                """,
                 default: Membrane.Time.seconds(10)
               ],
               start_composing_strategy: [
                 spec: :on_init | :on_message,
-                description:
-                  "Specifies when VideoCompositor starts composing frames. In `:on_message` strategy, `:start_composing` message have to be send to start composing.",
+                description: """
+                Specifies when VideoCompositor starts composing frames.
+                In `:on_message` strategy, `:start_composing` message have to be send to start composing.
+                """,
                 default: :on_init
               ],
               vc_server_port_number: [
-                spec: non_neg_integer(),
-                description:
-                  "Port on which VC server should run. Port have to be unused. In case of running multiple VC elements, those values should be unique.",
-                default: 8001
+                spec: :inet.port_number() | :choose_at_random,
+                description: """
+                Port on which VC server should run.
+                The port has to be unused. In case of running multiple VC elements, those values should be unique.
+                """,
+                default: :choose_at_random
               ]
 
   def_input_pad :input,
     accepted_format: %Membrane.H264{alignment: :nalu, stream_structure: :annexb},
     availability: :on_request,
     options: [
       input_id: [
-        spec: input_id(),
-        description: "Input identifier."
+        spec: input_id()
       ]
     ]
 
   def_output_pad :output,
     accepted_format: %Membrane.H264{alignment: :nalu, stream_structure: :annexb},
     availability: :on_request,
     options: [
-      resolution: [
-        spec: Resolution.t(),
-        description: "Resolution of output stream."
-      ],
       output_id: [
-        spec: output_id(),
-        description: "Output identifier."
-      ],
-      encoder_preset: [
-        spec: encoder_preset(),
-        description:
-          "Preset for an encoder. See [FFmpeg docs](https://trac.ffmpeg.org/wiki/Encode/H.264#Preset) to learn more.",
-        default: :medium
+        spec: output_id()
       ]
     ]
 
   @impl true
   def handle_init(_ctx, opt) do
-    vc_port = opt.vc_server_port_number
-    :ok = start_vc_server(vc_port)
+    {[], opt}
+  end
+
+  @impl true
+  def handle_setup(_ctx, opt) do
+    vc_port =
+      case opt.vc_server_port_number do
+        :choose_at_random ->
+          {port_lower_bound, port_upper_bound} = opt.port_range
+
+          [port] =
+            port_lower_bound..port_upper_bound
+            |> Enum.take_random(1)
 
-    :ok = VcReq.init(opt.framerate, opt.stream_fallback_timeout, opt.init_web_renderer?, vc_port)
+          port
+
+        port when is_integer(port) ->
+          port
+      end
+
+    :ok = ServerRunner.start_vc_server(vc_port)
```

Comment:
We won't retry if the randomly chosen port is busy. Not sure how problematic this is.